Gems/Atom/Tools/AtomToolsFramework/Code/Source/Util/Util.cpp

```diff
@@ -661,6 +662,197 @@ namespace AtomToolsFramework
         return paths;
     }
 
+    AZStd::string GetAbsolutePathForSourceAsset(const AZStd::string& path)
+    {
+        bool found = false;
+        AZ::Data::AssetInfo sourceInfo;
+        AZStd::string watchFolder;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            found, &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath, path.c_str(), sourceInfo, watchFolder);
+
+        if (found)
+        {
+            AZStd::string result;
+            if (AzFramework::StringFunc::Path::ConstructFull(watchFolder.c_str(), sourceInfo.m_relativePath.c_str(), result, true))
+            {
+                return result;
+            }
+        }
+
+        return path;
```

Comment:
AZ::StringFunc::Path::ConstructFull should be avoided and instead AZ::IO::Path should be used
```suggestion
            AZ::IO::Path result = AZ::IO::Path(watchFolder) / sourceInfo.m_relativePath;
            if (!result.empty())
            {
                return result.Native();
            }
        }

        return path;
```